CMakeLists.txt

```
@@ -167,7 +167,15 @@ FIND_PACKAGE( SIO QUIET )
 
 IF( NOT SIO_FOUND )
   MESSAGE( STATUS "SIO not found on your system. Using builtin sio" )
-  ADD_SUBDIRECTORY( sio )
+  include(FetchContent)
+  FetchContent_Declare(sio_extern
+  GIT_REPOSITORY https://github.com/iLCSoft/SIO.git
+  GIT_TAG v00-01
+  GIT_SHALLOW 1
+#  FIND_PACKAGE_ARGS
+)
```

```suggestion
  )

  get_target_property(SIO_VERSION SIO::sio VERSION)
  get_target_property(SIO_INCLUDE_DIRS SIO::sio INCLUDE_DIRECTORIES)
```

Just saw that these are use in a print out just here below and with FetchContent they are empty. This should hopefully also fix the ROOT dictionary issue.

---

Yes, sorry for delay, got only now around to it. Workaround works; I had previously not generated the rootDicts so I did not see the problem. It naturally is independent of the version. For the moment it should be okay but using root_generate_dictionary from the modern root cmake instead of the self made macros in MacroRootDict.cmake, the relevant target and MODULE option is I guess the way to go:
https://cliutils.gitlab.io/modern-cmake/chapters/packages/ROOT.html
https://github.com/root-project/root/issues/8308

---

No worries. Thanks for putting in the work :)

Yeah, switching to a more modern approach here has been on my list for quite some time, but haven't yet had time to come around to that.